hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/handler/OpenRegionHandler.java

```diff
@@ -277,6 +280,10 @@ Throwable getException() {
   /** Returns Instance of HRegion if successful open else null. */
   private HRegion openRegion() {
     HRegion region = null;
+    boolean compactionEnabled =
+      ((HRegionServer) server).getCompactSplitThread().isCompactionsEnabled();
+    this.server.getConfiguration().setBoolean(HBASE_REGION_SERVER_ENABLE_COMPACTION,
```

Comment:
Why change the global configuration here?

Reply:
We want to prefetch blocks for links/refs only if compaction is disabled, but the prefetch happens in the HFileReader, where we don't access to the CompactSplitThread, so saving the state of compaction switch in the config for HFileReader deciding if prefetch should go ahead.